src/librustc_typeck/check/mod.rs

```diff
@@ -954,7 +954,13 @@ fn diagnostic_only_typeck_tables_of<'tcx>(
     tcx: TyCtxt<'tcx>,
     def_id: DefId,
 ) -> &ty::TypeckTables<'tcx> {
-    let fallback = move || tcx.types.err;
+    let fallback = move || {
+        assert!(def_id.is_local());
+        let span =
+            tcx.hir().as_local_hir_id(def_id).map(|id| tcx.hir().span(id)).unwrap_or(DUMMY_SP);
```

Comment:
I don't think this can fail for local `DefId`s, so you can unwrap instead of `map`